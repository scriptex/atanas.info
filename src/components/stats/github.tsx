import * as React from 'react';
import ReactGitHubCalendar from 'react-ts-github-calendar';

import github from '../../data/github-insights.json';

import { StatsEntry, GithubSkyline } from '..';
import { GeneralInsight, YEARS } from './utils';
import { formatDate, isPrerendering } from '../../scripts/shared';

export const GithubStats: React.FC = () => {
	const [current, setCurrent] = React.useState(-1);
	const { error, general, calendar, updated, repositories } = github;

	if (error) {
		return (
			<div className="c-section__entry c-section__entry--no-background">
				<div className="o-shell">
					<h3>Github profile statistics</h3>

					<p>Failed fetching data from Github. Please check again in 4 hours.</p>
				</div>
			</div>
		);
	}

	const calendarDates = Object.keys(calendar).reduce((result: string[], key: string) => {
		if (Object.keys((calendar as any)[key]).length === 0) {
			return result;
		}

		return [...result, key];
	}, []);

	const blocks: GeneralInsight[] = [
		{
			title: 'Used languages',
			value: repositories
				.reduce((result: any[], repo: any) => Array.from(new Set([...result, repo.language])), [])
				.filter(Boolean)
				.join(', ')
		},
		{ title: 'Followers', value: general.followers },
		{ title: 'Following', value: general.following },
		{ title: 'Joined date', value: formatDate(general.createdAt) },
		{ title: 'Updated at', value: formatDate(general.updatedAt) },
		{ title: 'Last active', value: formatDate(calendarDates.reverse()[0]) },
		{ title: 'Total repositories', value: repositories.length },
		{ title: 'Private repositories', value: general.privateRepos },
		{ title: 'Public repositories', value: general.publicRepos },
		{
			title: 'Organizations repositories',
			value: repositories.length - general.publicRepos - general.privateRepos
		},
		{ title: 'Total gists', value: general.privateGists + general.publicGists },
		{ title: 'Private gists', value: general.privateGists },
		{ title: 'Public gists', value: general.publicGists },
		{
			title: 'Total contributions',
			value: repositories.reduce((result: number, repo: any) => {
				const contributor = repo.contributions.find(({ user }: { user: string }) => {
					user = user.toLowerCase();

					return user === 'scriptex' || user.includes('bot');
				});

				if (!contributor) {
					return result;
				}

				return result + contributor.count;
			}, 0)
		},
		{
			title: 'Total stars',
			value: repositories.reduce((result: number, repo: any) => result + repo.stargazers, 0)
		},
		{
			title: 'Total issues',
			value: repositories.reduce((result: number, repo: any) => result + repo.issues, 0)
		}
	];

	return (
		<>
			<StatsEntry data={blocks} title="Github profile statistics" />

			<div className="c-section__entry">
				<small className="c-section__stamp">Last updated: {formatDate(updated, 'dd MMM yyyy HH:mm:ss')}</small>

				<div className="o-shell">
					<h3>Github contributions calendar</h3>

					<div className="c-calendar__outer">
						<div className="c-calendar c-calendar--github">
							{isPrerendering ? null : <ReactGitHubCalendar tooltips userName="scriptex" />}
						</div>
					</div>

					<div className="c-skyline">
						<nav className="c-skyline__nav">
							<h4>
								Previous years Github contributions <br />
								<small>(requires WebGL)</small>
							</h4>

							<ul>
								{YEARS.map((year: string, index: number) => (
									<li key={index} className={current === index ? 'current' : undefined}>
										<button onClick={() => setCurrent(index)} className="c-btn c-btn--small">
											{year}
										</button>
									</li>
								))}
							</ul>
						</nav>

						{YEARS.map((year: string, index: number) =>
							index === current ? <GithubSkyline key={index} file={`${year}.stl`} index={index} /> : null
						)}
					</div>
				</div>
			</div>
		</>
	);
};

export default GithubStats;
