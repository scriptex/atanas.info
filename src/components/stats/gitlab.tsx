import * as React from 'react';
import { Link } from 'react-router-dom';
import GitlabCalendar from 'gitlab-calendar';

import gitlab from '../../data/gitlab-insights.json';
import gitlabCalendarData from '../../data/gitlab-calendar.json';

import { Routes } from '../../data/routes';
import { GeneralInsight } from './utils';
import { Ref, formatDate } from '../../scripts/shared';
import { sectionStatsProps } from '.';
import { Section, StatsEntry, StatsError } from '..';
import { GitlabInsights, GitlabRepository } from '../../scripts/types';

export const GitlabStats: React.FC = () => {
	const { error, general, calendar, updated, repositories }: GitlabInsights = gitlab;
	const calendarPlaceholder1: Ref<HTMLDivElement> = React.useRef(null);
	const calendarPlaceholder2: Ref<HTMLDivElement> = React.useRef(null);

	const blocks: GeneralInsight[] =
		!!repositories && !!general && !!calendar
			? [
					{
						title: 'Used languages',
						value: repositories
							.reduce(
								(result: string[], repo: GitlabRepository) =>
									Array.from(new Set([...result, ...Object.keys(repo.languages)])),
								[]
							)
							.filter(Boolean)
							.join(', ')
					},

					{ title: 'Joined date', value: formatDate(general.createdAt) },
					{ title: 'Last active', value: formatDate(general.updatedAt) },
					{
						title: 'Last year contributions',
						value: Object.values<number>(calendar).reduce(
							(total: number, value: number) => total + value,
							0
						)
					},
					{ title: 'Total repositories', value: repositories.length },
					{
						title: 'Public repositories',
						value: repositories.filter((r: GitlabRepository) => !r.private).length
					},
					{
						title: 'Private repositories',
						value: repositories.filter((r: GitlabRepository) => r.private).length
					},
					{
						title: 'Personal repositories',
						value: repositories.filter((r: GitlabRepository) => r.owner === 'scriptex').length
					},
					{
						title: 'Organizations repositories',
						value: repositories.filter((r: GitlabRepository) => r.owner !== 'scriptex').length
					},
					{
						title: 'Total stars',
						value: repositories.reduce(
							(result: number, repo: GitlabRepository) => result + repo.stargazers,
							0
						)
					},
					{
						title: 'Total issues',
						value: repositories.reduce(
							(result: number, repo: GitlabRepository) => result + (repo.issues || 0),
							0
						)
					}
			  ]
			: [];

	React.useEffect(() => {
		if (calendarPlaceholder1.current && !!calendar) {
			new GitlabCalendar(calendarPlaceholder1.current, calendar, {});
		}

		if (calendarPlaceholder2.current) {
			new GitlabCalendar(calendarPlaceholder2.current, gitlabCalendarData, {});
		}
	}, []);

	if (error || blocks.length === 0) {
		return <StatsError network="Gitlab" />;
	}

	return (
		<Section
			{...sectionStatsProps}
			actions={
				<Link to={Routes.STATS} className="c-btn">
					Go back
				</Link>
			}
			hasShell={false}
		>
			<StatsEntry data={blocks} title="Gitlab profile statistics" />

			<div className="c-section__entry">
				<small className="c-section__stamp">Last updated: {formatDate(updated, 'dd MMM yyyy HH:mm:ss')}</small>

				<div className="o-shell">
					<h3>Gitlab contributions calendars</h3>

					<div className="c-calendar__outer">
						<div className="c-calendar c-calendar--gitlab">
							<div className="c-calendar__entry">
								<h4>Public Gitlab profile</h4>

								<div ref={calendarPlaceholder1} />
							</div>

							<div className="c-calendar__entry">
								<h4>Private Gitlab profile</h4>

								<div ref={calendarPlaceholder2} />
							</div>
						</div>
					</div>
				</div>
			</div>
		</Section>
	);
};

export default GitlabStats;
