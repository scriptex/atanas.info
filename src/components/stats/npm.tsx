import * as React from 'react';
import { Link } from 'react-router-dom';

import npmStats from '../../data/npm-stats.json';
import { Routes } from '../../data/routes';
import { sectionStatsProps } from '.';
import { Section, ExternalLink } from '..';

interface Package {
	name: string;
	version: string;
	description: string;
	license: string;
	homepage: string;
	author: string;
	downloads: number;
}

interface Props {
	data: Record<string, Package>;
}

export const Packages: React.FC<Readonly<Props>> = ({ data }: Props) => (
	<div className="o-grid c-packages">
		{Object.keys(data).map((key: string, index: number) => {
			const item = data[key];
			const authors = item.author.split(',');

			return (
				<div className="o-grid__item xs-12 sm-6 md-4 lg-4" key={index}>
					<ExternalLink href={item.homepage} className="c-package">
						<h4>{item.name}</h4>

						<p>{item.description}</p>

						<ul>
							<li>
								Version <strong>{item.version}</strong>
							</li>

							<li>
								Downloads: <strong>{item.downloads}</strong>
							</li>

							<li>
								{authors.length > 1 ? 'Authors: ' : 'Author: '}

								<strong>{item.author}</strong>
							</li>
						</ul>
					</ExternalLink>
				</div>
			);
		})}
	</div>
);

export const NPMStats: React.FC = () => {
	const error: boolean = (npmStats as any)?.error;

	if (!npmStats || Object.keys(npmStats).length === 0 || error) {
		return null;
	}

	const { sum = null, ...packages } = npmStats;

	if (sum === null) {
		return null;
	}

	return (
		<Section
			{...sectionStatsProps}
			actions={
				<Link to={Routes.STATS} className="c-btn">
					Go back
				</Link>
			}
			hasShell={false}
		>
			<div className="c-section__entry c-section__entry--no-background c-section__entry--alt">
				<div className="o-shell">
					<h3>Statistics for NPM packages</h3>

					<h6>
						Total downloads: <strong>{sum}</strong>
					</h6>

					<Packages data={packages} />
				</div>
			</div>
		</Section>
	);
};

export default NPMStats;
