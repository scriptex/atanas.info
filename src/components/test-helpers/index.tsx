import * as React from 'react';
import { act } from 'react-dom/test-utils';
import { render, RenderResult, waitFor } from '@testing-library/react';

import * as shared from '../../scripts/shared';

export const mockUseEffect = (): void => {
	let useEffect: jest.Mock<React.EffectCallback>;

	const mockedUseEffect = () => {
		useEffect.mockImplementationOnce((f: () => React.EffectCallback) => f());
	};

	beforeEach(() => {
		useEffect = jest.spyOn(React, 'useEffect') as unknown as jest.Mock<React.EffectCallback>;

		mockedUseEffect();
	});
};

export const snapshotTest = (
	Component: React.FC<Readonly<React.ComponentProps<any>>>,
	buttonSelector?: string
): void => {
	const { name } = Component;

	jest.spyOn(shared, 'formatDate');

	(shared.formatDate as jest.Mock).mockImplementation(() => 'Mock date');

	mockUseEffect();

	afterAll(() => {
		jest.clearAllMocks();
	});

	describe(`${name} component`, () => {
		it(`Should render the ${name} component`, async () => {
			let result!: RenderResult;

			await act(async () => {
				result = await waitFor(() => render(<Component />));
			});

			const { asFragment, container } = result;

			expect(asFragment()).toMatchSnapshot();

			if (buttonSelector) {
				act(() => {
					container.querySelectorAll<HTMLElement>(buttonSelector).forEach(button => {
						if (!button.hasAttribute('download')) {
							button.click();

							expect(asFragment()).toMatchSnapshot();
						}
					});
				});
			}
		});

		it(`Should unmount the ${name} component`, async () => {
			let result!: RenderResult;

			await act(async () => {
				result = await waitFor(() => render(<Component />));
			});

			const { unmount, asFragment } = result;

			unmount();

			expect(asFragment()).toMatchSnapshot();
		});
	});
};
