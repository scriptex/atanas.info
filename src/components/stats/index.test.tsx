import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { Stats } from '.';
import { NPMStats } from './npm';
import { GitlabStats } from './gitlab';
import { GithubStats } from './github';
import { snapshotTest } from '../test-helpers';

jest.mock('../../data/github-insights.json', () => ({
	__esModule: true,
	default: {
		error: false,
		general: {
			publicRepos: 47,
			privateRepos: 5,
			publicGists: 20,
			privateGists: 8,
			followers: 50,
			following: 154,
			createdAt: '2013-06-03T18:11:48Z',
			updatedAt: '2020-06-23T10:52:09Z'
		},
		calendar: {
			'2019-06-23': {
				count: 10,
				color: '#c6e48b'
			},
			'2019-06-24': {
				count: 9,
				color: '#c6e48b'
			}
		},
		repositories: [
			{
				name: 'atanas.info',
				private: false,
				fork: false,
				createdAt: '2017-01-24T10:26:06Z',
				updated_at: '2020-06-23T14:42:49Z',
				size: 2042,
				stargazers: 4,
				watchers: 4,
				language: 'Javascript',
				issues: 1,
				contributions: [
					{
						user: 'scriptex',
						count: 301
					}
				]
			},
			{
				name: 'test',
				private: true,
				fork: false,
				createdAt: '2017-01-24T10:26:06Z',
				updated_at: '2020-06-23T14:42:49Z',
				size: 1024,
				stargazers: 0,
				watchers: 0,
				language: 'Typescript',
				issues: 2,
				contributions: [
					{
						user: 'test',
						count: 200
					}
				]
			}
		],
		updated: '2020-11-26T14:14:22.858Z'
	}
}));

jest.mock('../../data/gitlab-insights.json', () => ({
	__esModule: true,
	default: {
		error: false,
		general: {
			repos: 101,
			createdAt: '2018-01-03T11:47:34.435Z',
			updatedAt: '2020-06-24T13:19:32.897Z'
		},
		calendar: {
			'2020-03-06': 21,
			'2019-11-07': 34,
			'2019-10-26': 50
		},
		repositories: [
			{
				name: 'atanas.info',
				private: true,
				fork: 0,
				createdAt: '2019-08-14T06:13:34.958Z',
				updated_at: '2019-08-14T06:13:34.958Z',
				size: 18926796,
				stargazers: 0,
				languages: {
					Javascript: 47.72,
					HTML: 26.07,
					CSS: 25.79,
					PHP: 0.41,
					Shell: 0.01
				},
				issues: 0
			}
		],
		updated: '2020-11-26T14:14:22.858Z'
	}
}));

jest.mock('../../data/npm-stats.json', () => ({
	__esModule: true,
	default: {
		sum: 591668,
		package1: {
			name: 'package1',
			version: '1.0.0',
			description: 'Description 1',
			license: 'MIT',
			homepage: 'https://example.com',
			author: 'author',
			downloads: 0
		},
		package2: {
			name: 'package2',
			version: '1.0.0',
			description: 'Description 2',
			license: 'MIT',
			homepage: 'https://example.com',
			author: 'author',
			downloads: 0
		}
	}
}));

jest.mock('react-ts-github-calendar', () => ({
	__esModule: true,
	default: () => <div>Mocked Github Calendar</div>
}));

afterAll(() => {
	jest.resetAllMocks();
});

const StatsComponent: React.FC = () => (
	<Router>
		<Stats />
	</Router>
);

const GithubStatsComponent: React.FC = () => (
	<Router>
		<GithubStats />
	</Router>
);

const GitlabStatsComponent: React.FC = () => (
	<Router>
		<GitlabStats />
	</Router>
);

const NPMStatsComponent: React.FC = () => (
	<Router>
		<NPMStats />
	</Router>
);

[StatsComponent, GithubStatsComponent, GitlabStatsComponent, NPMStatsComponent].forEach(Component =>
	snapshotTest(Component)
);
