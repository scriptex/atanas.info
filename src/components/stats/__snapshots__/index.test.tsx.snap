// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubStatsComponent component Should render the GithubStatsComponent component 1`] = `
<DocumentFragment>
  <main
    class="o-main"
  >
    <section
      class="c-section c-section--stats"
      id="stats"
    >
      <header
        class="c-section__header"
      >
        <div
          class="o-shell"
        >
          <div
            class="c-section__animation"
            style="width: 150px; height: 150px; margin-left: auto; margin-right: auto;"
          />
          <h2>
            Stats
          </h2>
        </div>
      </header>
      <div
        class="c-section__entry c-section__entry--no-background"
      >
        <div
          class="o-shell"
        >
          <h3>
            Github profile statistics
          </h3>
          <ul
            class="c-section__list"
          >
            <li>
              <span>
                Used languages:
              </span>
              <strong>
                Javascript, Typescript
              </strong>
            </li>
            <li>
              <span>
                Followers:
              </span>
              <strong>
                50
              </strong>
            </li>
            <li>
              <span>
                Following:
              </span>
              <strong>
                154
              </strong>
            </li>
            <li>
              <span>
                Joined date:
              </span>
              <strong>
                Mock date
              </strong>
            </li>
            <li>
              <span>
                Updated at:
              </span>
              <strong>
                Mock date
              </strong>
            </li>
            <li>
              <span>
                Last active:
              </span>
              <strong>
                Mock date
              </strong>
            </li>
            <li>
              <span>
                Total repositories:
              </span>
              <strong>
                2
              </strong>
            </li>
            <li>
              <span>
                Private repositories:
              </span>
              <strong>
                5
              </strong>
            </li>
            <li>
              <span>
                Public repositories:
              </span>
              <strong>
                47
              </strong>
            </li>
            <li>
              <span>
                Organizations repositories:
              </span>
              <strong>
                -50
              </strong>
            </li>
            <li>
              <span>
                Total gists:
              </span>
              <strong>
                28
              </strong>
            </li>
            <li>
              <span>
                Private gists:
              </span>
              <strong>
                8
              </strong>
            </li>
            <li>
              <span>
                Public gists:
              </span>
              <strong>
                20
              </strong>
            </li>
            <li>
              <span>
                Total contributions:
              </span>
              <strong>
                301
              </strong>
            </li>
            <li>
              <span>
                Total stars:
              </span>
              <strong>
                4
              </strong>
            </li>
            <li>
              <span>
                Total issues:
              </span>
              <strong>
                3
              </strong>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="c-section__entry"
      >
        <small
          class="c-section__stamp"
        >
          Last updated: Mock date
        </small>
        <div
          class="o-shell"
        >
          <h3>
            Github contributions calendar
          </h3>
          <div
            class="c-calendar__outer"
          >
            <div
              class="c-calendar c-calendar--github"
            >
              <div>
                Mocked Github Calendar
              </div>
            </div>
          </div>
          <div
            class="c-skyline"
          >
            <nav
              class="c-skyline__nav"
            >
              <h4>
                Previous years Github contributions 
                <br />
                <small>
                  (requires WebGL)
                </small>
              </h4>
              <ul>
                <li>
                  <button
                    class="c-btn c-btn--small"
                  >
                    2013
                  </button>
                </li>
                <li>
                  <button
                    class="c-btn c-btn--small"
                  >
                    2014
                  </button>
                </li>
                <li>
                  <button
                    class="c-btn c-btn--small"
                  >
                    2015
                  </button>
                </li>
                <li>
                  <button
                    class="c-btn c-btn--small"
                  >
                    2016
                  </button>
                </li>
                <li>
                  <button
                    class="c-btn c-btn--small"
                  >
                    2017
                  </button>
                </li>
                <li>
                  <button
                    class="c-btn c-btn--small"
                  >
                    2018
                  </button>
                </li>
                <li>
                  <button
                    class="c-btn c-btn--small"
                  >
                    2019
                  </button>
                </li>
                <li>
                  <button
                    class="c-btn c-btn--small"
                  >
                    2020
                  </button>
                </li>
                <li>
                  <button
                    class="c-btn c-btn--small"
                  >
                    2021
                  </button>
                </li>
                <li>
                  <button
                    class="c-btn c-btn--small"
                  >
                    2022
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
      <div
        class="c-section__actions"
      >
        <button
          class="c-btn"
          type="button"
        >
          Get in touch
        </button>
        <ul
          class="c-section__actions-list"
        >
          <li>
            <button
              class="c-btn c-section__actions-close"
              type="button"
            >
              Close
            </button>
          </li>
          <li>
            <button
              class="c-btn"
              type="button"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-email"
                />
              </svg>
              <span>
                Contact me
              </span>
              <strong>
                directly
              </strong>
            </button>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://app.usebraintrust.com/talent/782/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-braintrust"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Braintrust
              </strong>
              <svg
                class="c-svg-external-link"
              >
                <use
                  xlink:href="#svg-external-link"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://www.toptal.com/resume/atanas-atanasov"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-toptal"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Toptal
              </strong>
              <svg
                class="c-svg-external-link"
              >
                <use
                  xlink:href="#svg-external-link"
                />
              </svg>
            </a>
          </li>
        </ul>
        <br
          class="visible-xs-block"
        />
        <a
          class="c-btn"
          href="/stats"
        >
          Go back
        </a>
      </div>
    </section>
  </main>
</DocumentFragment>
`;

exports[`GithubStatsComponent component Should unmount the GithubStatsComponent component 1`] = `<DocumentFragment />`;

exports[`GitlabStatsComponent component Should render the GitlabStatsComponent component 1`] = `
<DocumentFragment>
  <main
    class="o-main"
  >
    <section
      class="c-section c-section--stats"
      id="stats"
    >
      <header
        class="c-section__header"
      >
        <div
          class="o-shell"
        >
          <div
            class="c-section__animation"
            style="width: 150px; height: 150px; margin-left: auto; margin-right: auto;"
          />
          <h2>
            Stats
          </h2>
        </div>
      </header>
      <div
        class="c-section__entry c-section__entry--no-background"
      >
        <div
          class="o-shell"
        >
          <h3>
            Gitlab profile statistics
          </h3>
          <ul
            class="c-section__list"
          >
            <li>
              <span>
                Used languages:
              </span>
              <strong>
                Javascript, HTML, CSS, PHP, Shell
              </strong>
            </li>
            <li>
              <span>
                Joined date:
              </span>
              <strong>
                Mock date
              </strong>
            </li>
            <li>
              <span>
                Last active:
              </span>
              <strong>
                Mock date
              </strong>
            </li>
            <li>
              <span>
                Last year contributions:
              </span>
              <strong>
                105
              </strong>
            </li>
            <li>
              <span>
                Total repositories:
              </span>
              <strong>
                1
              </strong>
            </li>
            <li>
              <span>
                Public repositories:
              </span>
              <strong>
                0
              </strong>
            </li>
            <li>
              <span>
                Private repositories:
              </span>
              <strong>
                1
              </strong>
            </li>
            <li>
              <span>
                Personal repositories:
              </span>
              <strong>
                0
              </strong>
            </li>
            <li>
              <span>
                Organizations repositories:
              </span>
              <strong>
                1
              </strong>
            </li>
            <li>
              <span>
                Total stars:
              </span>
              <strong>
                0
              </strong>
            </li>
            <li>
              <span>
                Total issues:
              </span>
              <strong>
                0
              </strong>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="c-section__entry"
      >
        <small
          class="c-section__stamp"
        >
          Last updated: Mock date
        </small>
        <div
          class="o-shell"
        >
          <h3>
            Gitlab contributions calendars
          </h3>
          <div
            class="c-calendar__outer"
          >
            <div
              class="c-calendar c-calendar--gitlab"
            >
              <div
                class="c-calendar__entry"
              >
                <h4>
                  Public Gitlab profile
                </h4>
                <div />
              </div>
              <div
                class="c-calendar__entry"
              >
                <h4>
                  Private Gitlab profile
                </h4>
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c-section__actions"
      >
        <button
          class="c-btn"
          type="button"
        >
          Get in touch
        </button>
        <ul
          class="c-section__actions-list"
        >
          <li>
            <button
              class="c-btn c-section__actions-close"
              type="button"
            >
              Close
            </button>
          </li>
          <li>
            <button
              class="c-btn"
              type="button"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-email"
                />
              </svg>
              <span>
                Contact me
              </span>
              <strong>
                directly
              </strong>
            </button>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://app.usebraintrust.com/talent/782/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-braintrust"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Braintrust
              </strong>
              <svg
                class="c-svg-external-link"
              >
                <use
                  xlink:href="#svg-external-link"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://www.toptal.com/resume/atanas-atanasov"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-toptal"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Toptal
              </strong>
              <svg
                class="c-svg-external-link"
              >
                <use
                  xlink:href="#svg-external-link"
                />
              </svg>
            </a>
          </li>
        </ul>
        <br
          class="visible-xs-block"
        />
        <a
          class="c-btn"
          href="/stats"
        >
          Go back
        </a>
      </div>
    </section>
  </main>
</DocumentFragment>
`;

exports[`GitlabStatsComponent component Should unmount the GitlabStatsComponent component 1`] = `<DocumentFragment />`;

exports[`NPMStatsComponent component Should render the NPMStatsComponent component 1`] = `
<DocumentFragment>
  <main
    class="o-main"
  >
    <section
      class="c-section c-section--stats"
      id="stats"
    >
      <header
        class="c-section__header"
      >
        <div
          class="o-shell"
        >
          <div
            class="c-section__animation"
            style="width: 150px; height: 150px; margin-left: auto; margin-right: auto;"
          />
          <h2>
            Stats
          </h2>
        </div>
      </header>
      <div
        class="c-section__entry c-section__entry--no-background c-section__entry--alt"
      >
        <div
          class="o-shell"
        >
          <h3>
            Statistics for NPM packages
          </h3>
          <h6>
            Total downloads: 
            <strong>
              591668
            </strong>
          </h6>
          <div
            class="o-grid c-packages"
          >
            <div
              class="o-grid__item xs-12 sm-6 md-4 lg-4"
            >
              <a
                class="c-package"
                href="https://example.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <h4>
                  package1
                </h4>
                <p>
                  Description 1
                </p>
                <ul>
                  <li>
                    Version 
                    <strong>
                      1.0.0
                    </strong>
                  </li>
                  <li>
                    Downloads: 
                    <strong>
                      0
                    </strong>
                  </li>
                  <li>
                    Author: 
                    <strong>
                      author
                    </strong>
                  </li>
                </ul>
                <svg
                  class="c-svg-external-link"
                >
                  <use
                    xlink:href="#svg-external-link"
                  />
                </svg>
              </a>
            </div>
            <div
              class="o-grid__item xs-12 sm-6 md-4 lg-4"
            >
              <a
                class="c-package"
                href="https://example.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <h4>
                  package2
                </h4>
                <p>
                  Description 2
                </p>
                <ul>
                  <li>
                    Version 
                    <strong>
                      1.0.0
                    </strong>
                  </li>
                  <li>
                    Downloads: 
                    <strong>
                      0
                    </strong>
                  </li>
                  <li>
                    Author: 
                    <strong>
                      author
                    </strong>
                  </li>
                </ul>
                <svg
                  class="c-svg-external-link"
                >
                  <use
                    xlink:href="#svg-external-link"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c-section__actions"
      >
        <button
          class="c-btn"
          type="button"
        >
          Get in touch
        </button>
        <ul
          class="c-section__actions-list"
        >
          <li>
            <button
              class="c-btn c-section__actions-close"
              type="button"
            >
              Close
            </button>
          </li>
          <li>
            <button
              class="c-btn"
              type="button"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-email"
                />
              </svg>
              <span>
                Contact me
              </span>
              <strong>
                directly
              </strong>
            </button>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://app.usebraintrust.com/talent/782/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-braintrust"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Braintrust
              </strong>
              <svg
                class="c-svg-external-link"
              >
                <use
                  xlink:href="#svg-external-link"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              class="c-btn"
              href="https://www.toptal.com/resume/atanas-atanasov"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                class="c-section__actions-icon"
              >
                <use
                  xlink:href="#svg-toptal"
                />
              </svg>
              <span>
                Hire me on
              </span>
              <strong>
                Toptal
              </strong>
              <svg
                class="c-svg-external-link"
              >
                <use
                  xlink:href="#svg-external-link"
                />
              </svg>
            </a>
          </li>
        </ul>
        <br
          class="visible-xs-block"
        />
        <a
          class="c-btn"
          href="/stats"
        >
          Go back
        </a>
      </div>
    </section>
  </main>
</DocumentFragment>
`;

exports[`NPMStatsComponent component Should unmount the NPMStatsComponent component 1`] = `<DocumentFragment />`;

exports[`StatsComponent component Should render the StatsComponent component 1`] = `
<DocumentFragment>
  <main
    class="o-main"
  >
    <section
      class="c-section c-section--stats"
      id="stats"
    >
      <div
        class="o-shell"
      >
        <header
          class="c-section__header"
        >
          <div
            class="c-section__animation"
            style="width: 150px; height: 150px; margin-left: auto; margin-right: auto;"
          />
          <h2>
            Stats
          </h2>
        </header>
        <div
          class="c-section__grid o-grid"
        >
          <div
            class="o-grid__item xs-12 sm-6 md-4 lg-4 xl-4"
          >
            <a
              class="c-article-link fullsize-background"
              href="/stats/github"
              style="background-image: url(/images/temp/github.jpg);"
            >
              <strong>
                Github
              </strong>
            </a>
            <div
              class="c-loading"
            >
              <div
                class="c-loader"
              />
            </div>
          </div>
          <div
            class="o-grid__item xs-12 sm-6 md-4 lg-4 xl-4"
          >
            <a
              class="c-article-link fullsize-background"
              href="/stats/gitlab"
              style="background-image: url(/images/temp/gitlab.jpg);"
            >
              <strong>
                Gitlab
              </strong>
            </a>
            <div
              class="c-loading"
            >
              <div
                class="c-loader"
              />
            </div>
          </div>
          <div
            class="o-grid__item xs-12 sm-6 md-4 lg-4 xl-4"
          >
            <a
              class="c-article-link fullsize-background"
              href="/stats/npm"
              style="background-image: url(/images/temp/npm.jpg);"
            >
              <strong>
                NPM
              </strong>
            </a>
            <div
              class="c-loading"
            >
              <div
                class="c-loader"
              />
            </div>
          </div>
        </div>
        <div
          class="c-section__actions"
        >
          <button
            class="c-btn"
            type="button"
          >
            Get in touch
          </button>
          <ul
            class="c-section__actions-list"
          >
            <li>
              <button
                class="c-btn c-section__actions-close"
                type="button"
              >
                Close
              </button>
            </li>
            <li>
              <button
                class="c-btn"
                type="button"
              >
                <svg
                  class="c-section__actions-icon"
                >
                  <use
                    xlink:href="#svg-email"
                  />
                </svg>
                <span>
                  Contact me
                </span>
                <strong>
                  directly
                </strong>
              </button>
            </li>
            <li>
              <a
                class="c-btn"
                href="https://app.usebraintrust.com/talent/782/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  class="c-section__actions-icon"
                >
                  <use
                    xlink:href="#svg-braintrust"
                  />
                </svg>
                <span>
                  Hire me on
                </span>
                <strong>
                  Braintrust
                </strong>
                <svg
                  class="c-svg-external-link"
                >
                  <use
                    xlink:href="#svg-external-link"
                  />
                </svg>
              </a>
            </li>
            <li>
              <a
                class="c-btn"
                href="https://www.toptal.com/resume/atanas-atanasov"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  class="c-section__actions-icon"
                >
                  <use
                    xlink:href="#svg-toptal"
                  />
                </svg>
                <span>
                  Hire me on
                </span>
                <strong>
                  Toptal
                </strong>
                <svg
                  class="c-svg-external-link"
                >
                  <use
                    xlink:href="#svg-external-link"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </section>
  </main>
</DocumentFragment>
`;

exports[`StatsComponent component Should unmount the StatsComponent component 1`] = `<DocumentFragment />`;
